### PR TITLE
Modify RoslynCompilationService to generate portable pdbs

### LINKED ISSUE #4330: Modify RoslynCompilationService to generate portable pdbs
We have a check in this type to verify if the pdb writer component is registered. This does not work in x-plat scenarios (and might not work for CoreCLR on Windows). Instead we should switch to generating portable pdbs which should work in all environments.

### LINKED ISSUE #4430: MissingMethodException: Method not found at 1.0.0-rc2-*
Hi,

Referencing packages from RC2 (aspnetcidev endpoint) throws Missing Method Exception at runtime when:

```
  [HttpGet("~/signin")]
  public ActionResult SignIn(string returnUrl = null)
  {
      ViewBag.ReturnUrl = returnUrl;    
      return View("SignIn", HttpContext.GetExternalProviders());
  }
```

Current project.json:
```
{
  "dependencies": {
    "AspNet.Security.OAuth.Introspection": "1.0.0-alpha1-0114",
    "AspNet.Security.OAuth.Validation": "1.0.0-alpha1-0114",
    "AspNet.Security.OpenIdConnect.Server": "1.0.0-beta5-0691",
    "Microsoft.AspNetCore.Diagnostics": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Mvc": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Mvc.Razor": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Mvc.Core": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Mvc.Dnx": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Authentication.Cookies": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Authentication.Google": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Authentication.Twitter": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Hosting": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.IISPlatformHandler": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-rc2-*",
    "Microsoft.AspNetCore.StaticFiles": "1.0.0-rc2-*",
    "Microsoft.EntityFrameworkCore.InMemory": "1.0.0-rc2-*",
    "Microsoft.Extensions.Logging.Console": "1.0.0-rc2-*",
    "Microsoft.Extensions.Logging.Debug": "1.0.0-rc2-*"
  },
  "frameworks": {
    "net451": {
      "dependencies": {
        "Microsoft.AspNetCore.Owin": "1.0.0-rc2-*",
        "Microsoft.Owin.Security": "3.0.0",
        "NWebsec.Owin": "1.0.0"
      },
      "frameworkAssemblies": {
        "System.ComponentModel": "4.0.0.0"       
      }
    },
    "netstandardapp1.5": {
      "dependencies": {
        "NETStandard.Library": "1.5.0-*"
      },

      "imports": [
        "dnxcore50"
      ]
    }
  },
  "compilationOptions": {
      "emitEntryPoint": true,
      "preserveCompilationContext": true
  },
  "commands": {
      "web": "Mvc.Server"
  }
...
}
```
Stack trace:

```
An unhandled exception occurred while processing the request.

MissingMethodException: Method not found: 
'Void Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions..ctor(Microsoft.
CodeAnalysis.OutputKind, Boolean, System.String, System.String, System.String, 
System.Collections.Generic.IEnumerable`1<System.String>, 
Microsoft.CodeAnalysis.OptimizationLevel, Boolean, Boolean, System.String, System.String, 
System.Collections.Immutable.ImmutableArray`1<Byte>, System.Nullable`1<Boolean>, 
Microsoft.CodeAnalysis.Platform, Microsoft.CodeAnalysis.ReportDiagnostic, Int32, 
System.Collections.Generic.IEnumerable`1<System.Collections.Generic.KeyValuePair`2<
System.String,Microsoft.CodeAnalysis.ReportDiagnostic>>, Boolean, Boolean, 
Microsoft.CodeAnalysis.XmlReferenceResolver, Microsoft.CodeAnalysis.SourceReferenceResolver, 
Microsoft.CodeAnalysis.MetadataReferenceResolver, 
Microsoft.CodeAnalysis.AssemblyIdentityComparer, 
Microsoft.CodeAnalysis.StrongNameProvider, Boolean)'.

at Microsoft.AspNetCore.Mvc.Razor.RazorViewEngineOptions..ctor()


TargetInvocationException: An exception was thrown by the target of a call.

at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean 
noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& 
bNeedSecurityCheck)

``` 

I tried to add another versions of three packages too:

```
"Microsoft.AspNetCore.Mvc": "1.0.0-rc3-*",
"Microsoft.AspNetCore.Mvc.Razor": "1.0.0-rc3-*",
"Microsoft.AspNetCore.Mvc.Core": "1.0.0-rc3-*",
```

Nothing changed the exception throwed and it was working at 20270 build. I would like to freeze my build at latest RC2 version and to be waiting for RC2 release. It would be possible? Some mistake I made?

Current build: 20466.

Thanks

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #4330

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl

**Comment by Eilon** [comment]
FYI I fixed the bug number in the PR (it's 4330, not 4430), so you should fix your commit msg as well.

**Comment by Eilon** [comment]
Looks :shipit: to me. Do we want a Roslyn person to review as well?

**Comment by davidfowl** [comment]
This does make it impossible to debug razor views with older tooling.

:shipit:

**Comment by pranavkm** [comment]
Should we be keep the SymbolsUtility check and generate full pdbs if the pdbwriter is registered?

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -32,7 +32,6 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
     /// </summary>
     public class DefaultRoslynCompilationService : ICompilationService
     {
-        private readonly Lazy<bool> _supportsPdbGeneration = new Lazy<bool>(SymbolsUtility.SupportsSymbolsGeneration);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: :clap:

**Comment by davidfowl** [comment]
We need to merge this

**Pull request closed without merging**